util/file.py

```diff
@@ -94,6 +94,27 @@ def mkstempfname(suffix='', prefix='tmp', directory=None, text=False):
     return fn
 
 
+@contextlib.contextmanager
+def tempfname(*args, **kwargs):
+    '''Create a tempfile name on context entry, delete the file (if it exists) on context exit.'''
+    fn = mkstempfname(*args, **kwargs)
+    try:
+        yield fn
+    finally:
+        if os.path.isfile(fn): os.unlink(fn)
+
+
+@contextlib.contextmanager
+def tempfnames(suffixes, *args, **kwargs):
```

Comment:
Interesting, didn't know about ExitStack.  Still, it might end up as 2 lines vs 1 here.  Point of these small utils is to reduce bookkeeping clutter so that core algorithmic code becomes clearer, and for that purpose one-liner is better than two-liner.